src/mac_notifications/manager.py

```diff
@@ -48,12 +49,13 @@ class NotificationManager(metaclass=Singleton):
     """
 
     def __init__(self):
-        self._callback_queue: SimpleQueue = SimpleQueue()
+        self._callback_queue: SimpleQueue | None = None
         self._callback_executor_event: Event = Event()
         self._callback_executor_thread: CallbackExecutorThread | None = None
         self._callback_listener_process: NotificationProcess | None = None
         # Specify that once we stop our application, self.cleanup should run
         atexit.register(self.cleanup)
```

Comment:
@Jorricks why do you need both `atexit()` _and_ custom `SIGINT` handling that do the exact same thing (call `self.cleanup()`)? Seems like just `atexit()` should be sufficient to accomplish what you are going for?